src/components/richText/imageNode/ImageNode.tsx

```diff
@@ -0,0 +1,233 @@
+ */
+export class ImageNode extends DecoratorNode<JSX.Element> {
+  __src: string;
+  __altText: string;
+  __width: "inherit" | number;
+  __height: "inherit" | number;
+  __maxWidth: number;
+  __showCaption: boolean;
+  __caption: LexicalEditor;
+  // Captions cannot yet be used within editor cells
+  __captionsEnabled: boolean;
+
+  constructor(
+    src: string,
+    altText: string,
+    maxWidth: number,
+    width?: "inherit" | number,
+    height?: "inherit" | number,
+    showCaption?: boolean,
+    caption?: LexicalEditor,
+    captionsEnabled?: boolean,
+    key?: NodeKey
+  ) {
+    super(key);
+    this.__src = src;
+    this.__altText = altText;
+    this.__maxWidth = maxWidth;
+    this.__width = width || "inherit";
+    this.__height = height || "inherit";
+    this.__showCaption = showCaption || false;
+    this.__caption = caption || createEditor();
+    this.__captionsEnabled = captionsEnabled || captionsEnabled === undefined;
+  }
+
+  static getType(): string {
+    return "image";
+  }
+
+  static clone(node: ImageNode): ImageNode {
+    return new ImageNode(
+      node.__src,
+      node.__altText,
+      node.__maxWidth,
+      node.__width,
+      node.__height,
+      node.__showCaption,
+      node.__caption,
+      node.__captionsEnabled,
+      node.__key
+    );
+  }
+
+  getSrc(): string {
+    const latest = this.getLatest();
+    return latest.__src;
+  }
+
+  getAltText(): string {
+    const latest = this.getLatest();
+    return latest.__altText;
+  }
+
+  setWidthAndHeight(
+    width: "inherit" | number,
+    height: "inherit" | number
+  ): void {
+    const writable = this.getWritable();
+    writable.__width = width;
+    writable.__height = height;
+  }
+
+  setShowCaption(showCaption: boolean): void {
+    const writable = this.getWritable();
+    writable.__showCaption = showCaption;
+  }
+
+  /**
+   * Defines the JSON Serialization strategy for an {@link ImageNode}.
+   */
+  exportJSON(): SerializedImageNode {
+    return {
+      altText: this.getAltText(),
+      caption: this.__caption.toJSON(),
+      height: this.__height === "inherit" ? 0 : this.__height,
+      maxWidth: this.__maxWidth,
+      showCaption: this.__showCaption,
+      src: this.getSrc(),
+      type: "image",
+      version: 1,
+      width: this.__width === "inherit" ? 0 : this.__width,
+    };
+  }
+
+  /**
+   * Static constructor for creating an {@link ImageNode} from a JSON serialized Node.
+   */
+  static importJSON(serializedNode: SerializedImageNode): ImageNode {
+    const { altText, height, width, maxWidth, caption, src, showCaption } =
+      serializedNode;
+
+    const node: ImageNode = $applyNodeReplacement(
+      new ImageNode(src, altText, maxWidth, width, height, showCaption)
+    );
+    const nestedEditor = node.__caption;
+    const editorState = nestedEditor.parseEditorState(caption.editorState);
+    if (!editorState.isEmpty()) {
```

Comment:
for my own understanding, does this check ensure the node only displays a caption editor when a caption is present?

Reply:
Removed caption support! Turns out they weren't using it in Entity Edit.